src/deno_json/ts.rs

```diff
@@ -96,86 +95,55 @@ impl Serialize for IgnoredCompilerOptions {
   }
 }
 
-/// A static slice of all the compiler options that should be ignored that
-/// either have no effect on the compilation or would cause the emit to not work
-/// in Deno.
-pub const IGNORED_COMPILER_OPTIONS: &[&str] = &[
-  "allowImportingTsExtensions",
-  "allowSyntheticDefaultImports",
-  "allowUmdGlobalAccess",
-  "assumeChangesOnlyAffectDirectDependencies",
-  "baseUrl",
-  "build",
-  "charset",
-  "composite",
-  "declaration",
-  "declarationMap",
-  "diagnostics",
-  "disableSizeLimit",
-  "downlevelIteration",
-  "emitBOM",
-  "emitDeclarationOnly",
-  "esModuleInterop",
-  "extendedDiagnostics",
-  "forceConsistentCasingInFileNames",
-  "generateCpuProfile",
-  "help",
-  "importHelpers",
-  "incremental",
-  "init",
-  "inlineSourceMap",
-  "inlineSources",
-  "isolatedModules",
-  "listEmittedFiles",
-  "listFiles",
-  "mapRoot",
-  "maxNodeModuleJsDepth",
-  "module",
-  "moduleDetection",
-  "moduleResolution",
-  "newLine",
-  "noEmit",
-  "noEmitHelpers",
-  "noEmitOnError",
-  "noLib",
-  "noResolve",
-  "out",
-  "outDir",
-  "outFile",
-  "paths",
-  "preserveConstEnums",
-  "preserveSymlinks",
-  "preserveWatchOutput",
-  "pretty",
-  "project",
-  "reactNamespace",
-  "resolveJsonModule",
-  "rootDir",
-  "rootDirs",
-  "showConfig",
-  "skipDefaultLibCheck",
-  "skipLibCheck",
-  "sourceMap",
-  "sourceRoot",
-  "stripInternal",
-  "target",
-  "traceResolution",
-  "tsBuildInfoFile",
-  "typeRoots",
-  "useDefineForClassFields",
-  "version",
-  "watch",
-];
+/// A set of all the compiler options that should be allowed;
+static ALLOWED_COMPILER_OPTIONS: phf::Set<&'static str> = phf::phf_set! {
+  "allowUnreachableCode",
+  "allowUnusedLabels",
+  "strict",
+  "exactOptionalPropertyTypes",
+  "noFallthroughCasesInSwitch",
+  "noImplicitAny",
+  "noImplicitOverride",
+  "noImplicitReturns",
+  "noImplicitThis",
+  "noPropertyAccessFromIndexSignature",
+  "noUncheckedIndexedAccess",
+  "noUnusedLocals",
+  "noUnusedParameters",
+  "strictBindCallApply",
+  "strictFunctionTypes",
+  "strictNullChecks",
+  "strictPropertyInitialization",
+  "useUnknownInCatchVariables", // turn on by default
+  "types",
+  "checkJs",
+  "isolatedDeclarations",
+  "verbatimModuleSyntax",   // hook it up
+  "emitDecoratorMetadata",  // warn, unstable could be removed at any time
+  "experimentalDecorators", // warn, unstable could be removed at any time
```

Comment:
We're not warning for these yet - or do you want to do that in CLI?

Reply:
We're doing it in CLI.

Reply:
Let's do it in the cli